src/inmanta/app.py

```diff
@@ -55,31 +58,47 @@
 
 @command("server", help_msg="Start the inmanta server")
 def start_server(options):
-    io_loop = IOLoop.current()
-
     ibl = InmantaBootloader()
-    ibl.start()
 
-    try:
-        io_loop.start()
-    except KeyboardInterrupt:
-        IOLoop.current().stop()
-        ibl.stop()
+    def stop_server_signal_handler(signum, frame):
+        IOLoop.current().add_callback_from_signal(stop, ibl.stop)
```

Comment:
this code had me confused, could be a little more clear 
either

- give the stop function a clearer name (safe_shutdown_wrapper?)
- move the whole signal handling thing into a method that takes the stop function (ibl.stop I mean, not stop) as an argument and does all the signal setup?